options.go

```diff
@@ -51,15 +50,12 @@ func WithMiddleware(mw ...Middleware) ssh.Option {
 // WithHostKeyFile returns an ssh.Option that sets the path to the private.
 func WithHostKeyPath(path string) ssh.Option {
 	if _, err := os.Stat(path); os.IsNotExist(err) {
-		dir, f := filepath.Split(path)
-		n := strings.TrimSuffix(f, "_ed25519")
-		_, err := keygen.NewWithWrite(filepath.Join(dir, n), nil, keygen.Ed25519)
+		_, err := keygen.New(path+"_ed25519", keygen.WithKeyType(keygen.Ed25519), keygen.WithWrite())
```

Comment:
@aymanbagabas Looks like `keygen` trims the suffix so here we are double suffixing? I tried with the examples and it works like this.

Reply:
oof you're right that peace let me fix that and push a new patch release xD